File: Python/Day5/day5.py
#/* Pseudo Code
#	*  create an empty placeholder to collect number of parking space in the lots.
#	*  Allot parking space to each custom at available spaces.
#	* give options to display the number of available lots and the onces occupied.
#	* Let the users know if thereare available spaces left in the lot or not.
#	* Allow user to remove cars at request and display how many lots are left.
#*/

new_car = []
def main_menu():
    print("""
    Welcome to your Best CarPark...!
    List of menu....!
    1. Add Cars
    2. Show Available Car Space.
    0. Remove Cars.
    9. Exit.
      """)
    menup = int(input("Enter your choice: "))
    match menup:
        case 1:
            pack_car = input("Enter index to add: ")
            print(addCars(pack_car))
            main_menu()
        case 0:
            unpack_car = input("Enter index to unpack: ")
            print(remove_car(unpack_car))
            main_menu()
        case 9: 
        	exit_program()
        case _:
            print("Enter valid input: ")
            main_menu()


 

def addCars(pack_car):
	if (len(new_car) < 20):
		new_car.append(pack_car)
		return new_car
	else:
		return "Car Park Filled"
	
	
	
def remove_car(unpack_car):
	if (0 > len(new_car) > 21):
		new_car.remove(unpack_car)
		return new_car
	else:
		return "Car Space Empty"
		
	
		
def exit():
    print("Thank You For Choosing Us")
    exit()

main_menu()



 
	
	
	

File: Java/Day5/Day5.java
/* Pseudo Code
	*  create an empty placeholder to collect number of parking space in the lots.
	*  Allot parking space to each custom at available spaces.
	* give options to display the number of available lots and the onces occupied.
	* Let the users know if thereare available spaces left in the lot or not.
	* Allow user to remove cars at request and display how many lots are left.
*/


import java.util.ArrayList;
import java.util.Scanner;

public class CarParks {
    static ArrayList<Integer> newCar = new ArrayList<>();
    	static Scanner input = new Scanner(System.in);

    public static void main(String[] args) {
    
        		mainMenu();
    }

    public static void mainMenu() {
       String list = ("""
       			         Welcome to your Best Car Parking Lot...!
                List of menu....!
            		    1. Add items.
             		   0. Remove items.
           		     99. Show All Items
       			         9. Exit.
                """);
     System.out.print(list);
      int menup = input.nextInt();
       switch (menup){
       case 1:
        	System.out.print("Enter 1 to add: ");
        	int userInput = input.nextInt();
        	addCars(userInput);
         	mainMenu();
               	break;
        case 0:
        	System.out.print("Enter 0 to unpack: ");
        	int userInput2 = input.nextInt();
        	removeCar(userInput2);
        	mainMenu();
        
        case 9: 
        	exit();
         default:
         	System.out.print("Enter valid input: ");
           	mainMenu();
            	break;
           	 }
            }


    public static void addCars(int packCar) {
      if (packCar == 1){
		newCar.add(packCar);
		System.out.print(newCar);
		}
	else System.out.print("Car Park Filled");
    }



    public static void removeCar(int unpackCar) {
                     if (unpackCar == 0){
		newCar.remove(unpackCar);
		System.out.print(newCar);
		}
	else	System.out.print("Car Space Empty");
	}
    
    

    public static void  showAllItems() {
       System.out.print("Available Items:");
        for (int car : newCar) {
      	System.out.print(car);
      	 }
    }

    public static void exit() {
        System.out.print("Thank You For Choosing Us");
        System.exit(99);
    }
}

